### DISCUSSION TITLE
Can't send/fetch get request

### DISCUSSION THREAD

**Discussion by lethargistic** [vercel/community#4072] [Help] [answered]
### Summary

I'm new to Node, Express, Vercel and anything server side in general. I'm likely missing something obvious.

## Issue
I have Node.js (with Express) code that functions perfectly when run locally, it sends a get request from node.js back-end to front-end JS with res but when I deploy it on Vercel I get a 404 error in the console instead. I'm looking to use it to trasnfer a variable from back-end to front-end on a page.

When ran locally (as intended):
[functional](https://i.stack.imgur.com/SiKN0.png)
When deployed on Vercel: 
[disfunctional](https://i.stack.imgur.com/AAoT1.png)

Vercel seems to see both my static files and Node as indicated in the deplyment info, here's a screenshot:
[deployment](https://i.stack.imgur.com/Fwfai.png)

Here's my [project structure](https://i.stack.imgur.com/vCDSb.png) (image), everything seems to be in the right place.

## Possible solutions and attemps:
Since everything functions as intended when run locally the issue is likely caused by something in vercel.json. I've read the documentation throughly and I tried using rewrites in vercel.json but to no avail.
```
{
  "rewrites": [{ "source": "/api/(.*)", "destination": "/api" }]
}
```

As suggested by Nadeera3743 in this [Github issue](https://github.com/vercel/next.js/discussions/55150) I tried using Express to just display a get request withot static files and it works, **my issue is somehow related to the static files**.

I tried changing routes in Vercel.json and it seems to do something but it doesn't work nevertheless.

### Example

See code below, here's a link to the deployment:  https://test-vercel-8z708cz7e-maksiksking.vercel.app/

### Steps to Reproduce

Here is all of my code, there a 3 example requests and all of them function normally when run locally.

index.js

```
const express = require('express')
const {json} = require("express");

const index = express();

index.use(express.static('./public'));

let reqTest = '{\"name\":\"Test\"}'
console.log(JSON.parse(reqTest))
let d = JSON.parse(reqTest)

// Plain attempt (still works locally)
index.get('/test1', (req, res) => {
  res.json(d)
})

// Invalid json (route not specified in vercel.json)
index.get('/test2', (req, res) => {
  res.json(d)
})

// Seemingly hollow 404 response object (not parsed as json in scrip.js)
index.get('/test3', (req, res) => {
  res.json(d)
})

const port = process.env.PORT || 8080

index.listen(port, (err, res) => {
  if (err) {
    console.log(err)
    return res.status(500).send(err.message)
  } else {
    console.log('[INFO] Server Running on port:', port)
  }
})

```

scrip.js (I'm deeply sorry for my naming preferences)
```
document.addEventListener("DOMContentLoaded", function () {
    const bread = document.getElementById("hel")

    fetch('/test1')
        .then(d => d.json())
        .then(d => {
            console.log(d)
            bread.innerHTML = JSON.stringify(d)
        })
    fetch('/test2')
        .then(d => d.json())
        .then(d => {
            console.log(d)
            bread.innerHTML = JSON.stringify(d)
        })
    fetch('/test3')
        .then(d => {
            console.log(d)
            bread.innerHTML = JSON.stringify(d)
        })
});
```
vercel.json
```
{
  "version": 2,
  "builds": [
    {
      "src": "api/index.js",
      "use": "@vercel/node"
    },
    {
      "src": "public/**",
      "use": "@vercel/static"
    }
  ],
  "routes":[
    {
      "src": "/api/(.*)",
      "dest": "index.js"
    },
    {
      "src": "/",
      "dest": "public/index.html"
    },
    {
      "src": "/test1",
      "dest": "public/index.html"
    },
    {
      "src": "/(.+)",
      "dest": "public/$1"
    }
  ]
}
```
HTML
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
    <script src="javascripts/scrip.js"></script>
</head>
<body>
<h1 id="hel">replaceMe</h1>
<p>...other front-end stuff</p>

</body>
</html>
```

In addition to this my package.json includes the lines to run Node:
```
  "scripts": {
    "start": "node api/index.js",
    "server": "nodemon index.js"
  },
```

I've been stuck with this for a week, any help is well appreciated. I'm open to changing anything in the post and providing more info.

**Comment by amyegan** [comment]
Hi @maksiksking. There are a few structure and config changes I think you'll need to make. This guide has info to help you deploy an Express app to Vercel using Serverless Functions: https://vercel.com/guides/using-express-with-vercel

**Reply by lethargistic** [reply]
Hello @amyegan, I've used that exact guude and it functions, and works when I add my static files, but a fetch request from my front-end JavaScript still is a 404.

**Comment by lethargistic** [comment, marked as answer]
I solved it thanks to @Nadeera3784 in this thread:
https://github.com/vercel/next.js/discussions/55150
